### PR TITLE
meson: 1.2.3 -> 1.3.0

### PR DESCRIPTION
**Pull request by alyssais**
## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [x] aarch64-linux
  - [x] x86_64-darwin
  - [x] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by alyssais** [comment]
@jtojnar it looks like the PR for the patch removed here still hasn't been merged, but now conflicts — would you mind rebasing it?

**Comment by jtojnar** [comment]
I think the patch should not be needed any more. IIRC, this was meant to fix build of fwupd, which had been passing multiple variables to `define_variable`, when Meson suddenly started rejecting that, where it previously silently ignored it. I believe fwupd has since fixed the build script.

**Comment by alyssais** [comment]
Ah cool, okay.  Let's try.

@OfBorg build fwupd

**Comment by alyssais** [comment]
Built fwupd locally; seems fine!

**Comment by mweinelt** [comment]
I need this update to unblock meson-python on Python 3.12.

Specifically this patch: https://github.com/mesonbuild/meson/commit/20bcca39726803aba56fcc388332622d70605e1a

**Comment by alyssais** [comment]
I've done a big build with it and everything seemed fine.  I found https://github.com/mesonbuild/meson/issues/12585 but it's pretty niche (affects pkgsStatic + rust — I'm not sure we even have any packages using Meson and Rust that build statically in Nixpkgs).

**Comment by mweinelt** [comment]
> Specifically this patch: [mesonbuild/meson@20bcca3](https://github.com/mesonbuild/meson/commit/20bcca39726803aba56fcc388332622d70605e1a)

This one

**Comment by alyssais** [comment]
Ah, I see.  I misread as there being a patch for the issue I linked.  I think it's fine to go ahead with this — I'm very likely the only user of the affected behaviour, and my code that uses it is not in Nixpkgs, so I don't think it's right to block Nixpkgs on it.

File: pkgs/by-name/me/meson/package.nix [hunk 1]
```diff
@@ -66,15 +66,8 @@ python3.pkgs.buildPythonApplication rec {
...
-    # https://github.com/mesonbuild/meson/pull/10670
-    (fetchpatch {
-      url = "https://github.com/mesonbuild/meson/commit/d5252c5d4cf1c1931fef0c1c98dd66c000891d21.patch";
-      hash = "sha256-GiUNVul1N5Fl8mfqM7vA/r1FdKqImiDYLXMVDt77gvw=";
-      excludes = [
-        "docs/yaml/objects/dep.yaml"
-      ];
-    })
+    # https://github.com/mesonbuild/meson/pull/12587
+    ./007-darwin-case-sensitivity.patch
```
Comment (mweinelt) [inline pkgs/by-name/me/meson/package.nix:70]: The patch is not in the commit.

**Review by mweinelt — APPROVED** [review]

**Pull request merged**